src/tests/stress/pedsim/Ped_util/Regions.enc

```diff
@@ -0,0 +1,462 @@
+import Quad_tree
+
+fun regions(var agents : [Agent], boxes : int, steps : int) : unit
+  var box = new Tiling_box(agents, boxes)
+  box!move(steps)
+end
+fun link(a : Box, b : Box) : bool
+  val aa = a!link(b)
+  val bb = b!link(a)
+  val aaa = get(aa)
+  val bbb = get(bb)
+  (aaa && bbb)
+end
+active class Tiling_box
+  var boxes : [Box]
+  var futures : [Fut[bool]]
+  var futs_left : int
+  def init(agents : [Agent], n : int) : unit
+    match find_extreme(agents) with
+      case (xmax, ymax, xmin, ymin) =>
+        val dx = (xmax - xmin) / n
+        this.boxes = new [Box]( n * n )
+        this.futures = new [Fut[bool]](|this.boxes|)
+        if(n>1) then
+            var x = xmin
+            var y = ymin
+            var i = 0
+            var win = true
+            val dy = (ymax - ymin) / n
+            for xindex <- [0..n - 2] do
+              for yindex <- [0..n - 2] do
+                this.boxes(i) = new Box((x + dx - 1, y + dy - 1), (x, y))
+                if yindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+                end
+                if xindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n))
+                end
+                if xindex != 0 && yindex != 0 then
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n + 1))
+                  win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+                  ()
+                end
+                i = i + 1
+                y = y + dy
+              end
+              this.boxes(i) = new Box((x + dx - 1, ymax), (x, y))
+              if xindex != 0 then
+                win = win && link((this.boxes)(i), (this.boxes)(i - n))
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+              i = i + 1
+              x = x + dx
+              y = ymin
+            end
+            for yindex <- [0..n - 2] do
+              (this.boxes)(i) = new Box((xmax, y + dy - 1), (x, y))
+              if yindex != 0 then
+                win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+                win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+              end
+              win = win && link((this.boxes)(i), (this.boxes)(i - n + 1))
+              win = win && link((this.boxes)(i), (this.boxes)(i - n))
+              i = i + 1
+              y = y + dy
+            end
+            (this.boxes)(i) = new Box((xmax, ymax), (x, y))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n))
+            win = win && link((this.boxes)(i), (this.boxes)(i - 1))
+            win = win && link((this.boxes)(i), (this.boxes)(i - n - 1))
+        else
+            this.boxes(0) = new Box((xmax,ymax), (xmin,ymin))
+        end
+        for a <- agents do
+          for b <- this.boxes do
+            if get(b!add(a)) then
```

Comment:
This is method is only called in setup, which is NOT optimised at all, and not measured in performance measurements. So yea, it's serial.